lib/oracle.js

```diff
@@ -150,61 +179,93 @@ Oracle.prototype.executeSQL = function (sql, params, options, callback) {
   if (transaction && transaction.connection &&
     transaction.connector === this) {
     debug('Execute SQL within a transaction');
-    transaction.connection.execute(sql, params, function (err, data) {
-      if (err && self.settings.debug) {
-        self.debug(err);
-      }
-      if (self.settings.debug && data) {
-        self.debug("Result: %j", data);
-      }
-      callback(err, data);
-    });
+    transaction.connection.execute(sql, params,
+      {outFormat: oracle.OBJECT, autoCommit: false}, function(err, data) {
+        if (err && self.settings.debug) {
+          self.debug(err);
+        }
+        if (self.settings.debug && data) {
+          self.debug("Result: %j", data);
+        }
+        if (data && data.rows) {
+          data = data.rows;
+        }
+        callback(err, data);
+      });
     return;
   }
 
-  /*
-  self.pool.getConnection(function (err, connection) {
+  self.pool.getConnection(function(err, connection) {
     if (err) {
       callback && callback(err);
       return;
     }
-    console.log('open %d -> %d', open, ++open);
     if (self.settings.debug) {
-      debugConnection('Connection acquired: ', self.pool.getInfo());
+      self.debug('Connection acquired: ', self.pool);
     }
-    connection.execute(sql, params, function (err, data) {
-      if (err && self.settings.debug) {
-        self.debug(err);
-      }
-      if (self.settings.debug && data) {
-        self.debug("Result: %j", data);
-      }
-      if (log) {
-        log(sql, time);
-      }
-      connection.close(function(err) {
-        if (err) {
-          return callback(err);
+    connection.clientId = self.settings.clientId || 'LoopBack';
+    connection.module = self.settings.module || 'loopback-connector-oracle';
+    connection.action = self.settings.action || '';
+
+    connection.execute(sql, params,
+      {outFormat: oracle.OBJECT, autoCommit: true},
+      function(err, data) {
+        if (err && self.settings.debug) {
+          self.debug(err);
         }
-        console.log('close %d -> %d', open, --open);
-        if (self.settings.debug) {
-          debugConnection('Connection released: ', self.pool.getInfo());
+
+        if (!err && data) {
+          var lobs = [];
+          if (data.rows) {
+            data = data.rows;
+            if (self.settings.debug && data) {
+              self.debug("Result: %j", data);
+            }
+            if (Array.isArray(data)) {
+              for (var i = 0, n = data.length; i < n; i++) {
+                // Iterate through the rows
+                var row = data[i];
+                for (var k in row) {
+                  var val = row[k];
+                  if (val instanceof stream.Readable) {
+                    lobs.push({index: i, key: k, stream: val});
+                  }
```

Comment:
I am not quite familiar with stream and lob operations so I might be wrong, just wondering is it ok to discard the data if it's  `not instanceof steam.Readable`?